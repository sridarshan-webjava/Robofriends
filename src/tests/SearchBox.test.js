import { shallow } from "enzyme";
import React from "react";
import SearchBox from "../SearchBox";

let wrapper;
const mockSearchChangeFn = jest.fn();

beforeEach(() => {
  wrapper = shallow(<SearchBox searchChange={mockSearchChangeFn} />);
});

it("Test if the snapshot match", () => {
  expect(wrapper).toMatchSnapshot();
});

it("Test if input event function is triggered", () => {
  const input = wrapper.find("input");
  input.simulate("change");
  expect(mockSearchChangeFn).toBeCalled();
});
