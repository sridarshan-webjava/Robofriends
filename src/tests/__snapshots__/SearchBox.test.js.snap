// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the snapshot match 1`] = `ShallowWrapper {}`;
